src/apis/market.ts

```diff
@@ -12,6 +12,27 @@ import { cachedPublicClients } from '../constants/client'
 import { BOOK_VIEWER_ABI } from '../abis/core/book-viewer-abi'
 import { fetchIsOpened } from '../utils/open'
 import { fetchCurrency } from '../utils/currency'
+import { cachedSubgraph } from '../constants/subgraph'
+
+const fetchBook = async (chainId: CHAIN_IDS, bookId: string) => {
+  return cachedSubgraph[chainId]!.get<{
```

Comment:
q2; look dangerous pattern, how about building subgraph here if cached subgraph is undefined, or just throwing error?